apiserver/facades/client/client/client.go

```diff
@@ -180,9 +182,20 @@ func newFacade(ctx facade.Context) (*Client, error) {
 		return nil, errors.Trace(err)
 	}
 
-	modelCache, err := ctx.Controller().Model(modelUUID)
-	if err != nil {
-		return nil, errors.Trace(err)
+	var modelCache *cache.Model
+	if featureflag.Enabled(feature.Generations) {
```

Comment:
It might be better to make this check against the feature flags in controller config - that way the flag could be toggled without rebootstrapping. See `apiserver/facades/client/cloud/cloud.go` for an example (search for `feature.MultiCloud`).

Reply:
Updated and tested.